services/src/main/java/org/keycloak/protocol/saml/profile/ecp/authenticator/HttpBasicAuthenticator.java

```diff
@@ -39,8 +39,12 @@ public void authenticate(final AuthenticationFlowContext context) {
                 final boolean valid = context.getSession().userCredentialManager().isValid(realm, user, UserCredentialModel.password(password));
 
                 if (valid) {
-                    context.getClientSession().setAuthenticatedUser(user);
-                    context.success();
+                    if (user.isEnabled()) {
+                        context.getClientSession().setAuthenticatedUser(user);
+                        context.success();
+                    } else {
+                        userDisabledFailure(context, realm, user);
+                    }
```

Comment:
Yes, I believe this should be in a separate PR?
@pedroigor Can you confirm the way this should fail for disabled users is a proper fix?